spec/requests/providers/application_merits_task/opponent_organisations_controller_spec.rb

```diff
@@ -0,0 +1,161 @@
+          patch(
+            providers_legal_aid_application_opponent_organisation_path(legal_aid_application, opponent),
+            params: params.merge(button_clicked),
+          )
+        end
+
+        let!(:opponent) { create(:opponent, :for_organisation, legal_aid_application:, organisation_name: "Mid Bedfordshire Council", organisation_ccms_code: "GOVT", organisation_description: "Government Department") }
+
+        let(:params) do
+          {
+            application_merits_task_opponent: {
+              name: "Central Bedfordshire Council",
+              organisation_type_ccms_code: "LA",
+            },
+          }
+        end
+
+        let(:draft_button) { { draft_button: "Save as draft" } }
+        let(:button_clicked) { {} }
+
+        before do
+          allow(LegalFramework::MeritsTasksService).to receive(:call).with(legal_aid_application).and_return(smtl)
+          login_provider
+        end
+
+        it "amends the opponent opposable organisation with the values entered" do
+          expect { patch_organisation }.not_to change(::ApplicationMeritsTask::Organisation, :count)
+          opponent.opposable.reload
+          expect(opponent.opposable.name).to eql "Central Bedfordshire Council"
+          expect(opponent.opposable.ccms_code).to eql "LA"
+        end
+
+        it "sets the task to complete" do
+          patch_organisation
+          expect(legal_aid_application.legal_framework_merits_task_list.serialized_data).to match(/name: :opponent_name\n\s+dependencies: \*\d+\n\s+state: :complete/)
+        end
+
+        it "redirects to the has another opponent question" do
+          patch_organisation
+          expect(response).to redirect_to(providers_legal_aid_application_has_other_opponent_path(legal_aid_application))
+        end
+
+        context "when not authenticated" do
+          let(:login_provider) { nil }
+
+          before { patch_organisation }
+
+          it_behaves_like "a provider not authenticated"
+        end
+
+        context "when incomplete" do
+          let(:params) do
+            {
+              application_merits_task_opponent: {
+                name: "",
+                organisation_type_ccms_code: "",
+              },
+            }
+          end
+
+          it "renders show" do
+            patch_organisation
+            expect(response).to have_http_status(:ok)
+          end
+
+          it "contains error message" do
+            patch_organisation
+            expect(response.body).to include("govuk-error-summary")
+            .and include(html_compare("Enter the organisation's name"))
+            .and include(html_compare("Select the organisation's type"))
+          end
+
+          it "does not set the task to complete" do
+            patch_organisation
+            expect(legal_aid_application.legal_framework_merits_task_list.serialized_data).to match(/name: :opponent_name\n\s+dependencies: \*\d+\n\s+state: :not_started/)
```

Comment:
I added some merits task matchers a while back, you should be able to use (something like) 
```ruby
expect(legal_aid_application.legal_framework_merits_task_list).to have_task_in_state(:application, :opponent_name, :not_started)
```

Reply:
Thanks the merits task matchers are nice and  have made suggested change 👍